editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/provider/ToolBehaviourProvider.java

```diff
@@ -742,7 +742,13 @@ public int compare(IToolEntry o1, IToolEntry o2) {
 		ArrayList<IToolEntry> entries = new ArrayList<IToolEntry>();
 
 		// inject palette entries generated out of the component model file
-		String camelVersion = CamelUtils.getCurrentProjectCamelVersion();
+		CamelDesignEditor editor = null;
+		if (getDiagramTypeProvider() != null && 
+			getDiagramTypeProvider().getDiagramBehavior() != null &&
+			getDiagramTypeProvider().getDiagramBehavior().getDiagramContainer() != null) {
+			editor = (CamelDesignEditor)getDiagramTypeProvider().getDiagramBehavior().getDiagramContainer();
+		}
+		String camelVersion = editor != null ? CamelUtils.getCurrentProjectCamelVersion(editor) : CamelUtils.getCurrentProjectCamelVersion();
```

Comment:
the null case is handled in CamelUtils.getCurrentProjectCamelVersion but differently...
what do you think to move all the logic to retrieve the camel version in CamelUtils?